### PR TITLE
[osgearth] Update to version 3.4

### LINKED ISSUE #31532: [osgEarth] update to <version>3.4
### Library name

osgEarth

### New version number

3.4

### Other information that may be useful (release notes, etc...)

https://github.com/gwaldron/osgearth
https://github.com/gwaldron/osgearth/archive/refs/tags/osgearth-3.4.tar.gz

### PR DESCRIPTION
**Pull request by cbrl**
Fixes #31532

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/osgearth/find-package.patch [hunk 1]
```diff
@@ -92,21 +92,21 @@ index 36ea6c81f..8ed39eb24 100755
...
  ENDIF (WITH_EXTERNAL_TINYXML)
  
  # postfix settings for various configs
-@@ -300,6 +320,7 @@ IF(OSGEARTH_BUILD_SHARED_LIBS)
+@@ -307,6 +327,7 @@ IF(OSGEARTH_BUILD_SHARED_LIBS)
      SET(OSGEARTH_DYNAMIC_OR_STATIC "SHARED")
  ELSE()
      SET(OSGEARTH_DYNAMIC_OR_STATIC "STATIC")
-+    add_definitions(-DOSGEARTH_LIBRARY_STATIC)
++	add_definitions(-DOSGEARTH_LIBRARY_STATIC)
```
Comment (dg0yt) [inline ports/osgearth/find-package.patch:109]: (nice-to-have) If you do another push, can you fix the intentation in this file? There seem to be tabs now where there should be four spaces.

Comment (cbrl) [reply ports/osgearth/find-package.patch]: Must have been the auto-formatting from my editor. It's fixed now.

**Review by LilyWangLL — DISMISSED** [review]

**Comment by LilyWangLL** [comment]
Thanks for your update. Please also fix the following warning:
```
.\vcpkg.exe install osgearth[*]:x64-windows --binarysource=clear
Building osgearth[blend2d,core,default-features,tools]:x64-windows

......

-- Installing: E:/PRreview/osg/packages/osgearth_x64-windows/share/osgearth/copyright
-- Performing post-build validation
warning: The following EXEs were found in /bin or /debug/bin. EXEs are not valid distribution targets.

    E:\PRreview\osg\packages\osgearth_x64-windows\bin\osgearth_bakefeaturetiles.exe

warning: There should be no empty directories in E:\PRreview\osg\packages\osgearth_x64-windows. The following empty directories were found:

    E:\PRreview\osg\packages\osgearth_x64-windows\tools\osgearth\debug

warning: If a directory should be populated but is not, this might indicate an error in the portfile.
If the directories are not needed and their creation cannot be disabled, use something like this in the portfile to remove them:
    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/tools/osgearth/debug")
```
The first warning could be fixed by adding tool name `osgearth_bakefeaturetiles` to `vcpkg_copy_tools` in `portfile.cmake` Line 89. 
The next warning could be fixed by adding `file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/tools/osgearth/debug")` in `portfile.cmake`.

**Comment by LilyWangLL** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by LilyWangLL** [comment]
Usage test passed on `x64-windows`.

**Review by LilyWangLL — APPROVED** [review]

**Comment by LilyWangLL** [comment]
Features passed on `x64-windows`.

**Pull request merged**